### PR TITLE
Added IPC project files.

### PR DESCRIPTION
**Pull request by kollil**
#### Rationale
Created a new fb for the IPC Histology service request process project files.

### CODE REVIEW

**Review by Ohsudev — APPROVED** [review]

**Comment by labkey-jeckels** [comment]
The automated tests are hitting an error with onprc_ehr-20.418-20.419.sql:

```
ERROR ModuleUpgrader           2022-02-14T17:49:28,743           Module Upgrade : Failure during module upgrade
org.labkey.api.data.SqlScriptRunner$SqlScriptException: onprc_ehr-20.418-20.419.sql : DROP xxx IF EXISTS statements are prohibited since they're not supported on SQL Server 2012. Instead, use EXEC core.fn_dropifexists.
```

**Review by labkey-jeckels — CHANGES_REQUESTED** [review]
See my note about the test failures due to the schema upgrade script.

File: onprc_ehr/resources/web/onprc_ehr/model/sources/IPC_Child.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        Id: {
+            editable: false,
+            columnConfig: {
+                editable: false
+            }
+        },
+        date: {
+            inheritDateFromParent: true
+        }
+        // project: {
```
Comment (labkey-jeckels) [inline onprc_ehr/resources/web/onprc_ehr/model/sources/IPC_Child.js]: Can these commented sections be deleted?

File: onprc_ehr/src/org/labkey/onprc_ehr/dataentry/IPC_ServiceRequestDetailsFormSection.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+//package org.labkey.onprc_ehr.dataentry;
+//
+//import org.labkey.api.ehr.EHRService;
+//import org.labkey.api.ehr.dataentry.SimpleFormPanelSection;
+//import org.labkey.api.view.template.ClientDependency;
+//
+///**
+// * User: Kolli
+// * Date: 4/01/2020
+// * Time: 10:36 AM
```
Comment (labkey-jeckels) [inline onprc_ehr/src/org/labkey/onprc_ehr/dataentry/IPC_ServiceRequestDetailsFormSection.java:4]: Can this commented section be deleted?

File: onprc_ehr/src/org/labkey/onprc_ehr/dataentry/IPC_SlidePrintingFormSection.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+        if (defaultButtons.contains("COPYFROMSECTION"))
+        {
+            int idx = defaultButtons.indexOf("COPYFROMSECTION");
+            defaultButtons.remove("COPYFROMSECTION");
+            defaultButtons.add(idx, "IPC_COPYFROMSECTION");
+        }
+
+        return defaultButtons;
+    }
```
Comment (labkey-jeckels) [inline onprc_ehr/src/org/labkey/onprc_ehr/dataentry/IPC_SlidePrintingFormSection.java:44]: This method is repeated in many files. I recommend introducing a new superclass that handles this, and then extending that for each of the different sections.

**Review by labkey-jeckels — APPROVED** [review]
Some suggestions for cleanup, but your call on whether to tackle them now.

File: onprc_ehr/src/org/labkey/onprc_ehr/dataentry/IPC_SlidePrintingFormSection.java (hunk 3 above)
Comment (kollil) [reply onprc_ehr/src/org/labkey/onprc_ehr/dataentry/IPC_SlidePrintingFormSection.java:44]: That makes sense. I need to improvise the code in many places, it was written in 19.1 version and project got shelved for different reasons. The new management is willing to resume the development slowly depending on how much functionality is already implemented. So, I am trying to restore the code into 21.7 and created a  new FB for this project. Thank you always for your feedback @labkey-jeckels.

**Pull request closed without merging**